src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -44,22 +49,37 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs
 
         logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
 
-        this.trackedAddressBook = new TrackedAddressBook(addressBook, userPrefs.getShortcutSettings());
+        this.addressBook = new AddressBook(addressBook);
         this.userPrefs = new UserPrefs(userPrefs);
-        filteredPersons = new FilteredList<>(this.trackedAddressBook.getPersonList());
+        filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
         this.commandStringStash = new CommandStringStash();
         this.themeProperty = new ThemeProperty();
+        this.modelManagerStateList = new ArrayList<>();
+        this.currentStatePointer = -1; //starts from -1 because commit() will +1 later on
 
         // DoConnek Pro shows all patients on startup by default.
         updateFilteredPersonList(PersonType.PATIENT.getSearchPredicate());
 
         this.selectedPerson = filteredPersons.size() == 0 ? null : filteredPersons.get(0);
+        commit();
     }
 
     public ModelManager() {
         this(new AddressBook(), new UserPrefs());
     }
 
+    /**
+     * Resets this ModelManager's data to copy another ModelManager's
+     * @param toCopyFrom ModelManager to copy from
+     */
+    public void resetData(ReadOnlyModelManager toCopyFrom) {
+        this.addressBook = toCopyFrom.addressBook;
```

Comment:
This section is what is causing the error, the observers are attached to the objects, if you change the objects then the observers are no longer attached to the new objects.
You should use the corresponding set methods instead to mutate the objects directly (tried on my system, seems to be working) e.g.
```
        setAddressBook(toCopyFrom.addressBook);
        setUserPrefs(toCopyFrom.userPrefs);
        updateSelectedPerson(toCopyFrom.selectedPerson);
        updateFilteredPersonList((Predicate<Person>) toCopyFrom.filteredPersons.getPredicate());
        setShortcutSettings(toCopyFrom.userPrefs.getShortcutSettings());
        setTheme(toCopyFrom.themeProperty.getValue());
```
(I'm not sure if the typecast for `Predicate<Person>` will always be valid, maybe add a check for this)